assets/lokomotive-kubernetes/aws/flatcar-linux/kubernetes/controllers.tf

```diff
@@ -13,6 +13,70 @@ resource "aws_route53_record" "etcds" {
   records = [aws_instance.controllers[count.index].private_ip]
 }
 
+# IAM Policy
+resource "aws_iam_instance_profile" "csi-driver-instance-profile" {
+  count = var.set_csi_driver_iam_role ? 1 : 0
+  name  = "${var.cluster_name}-iprof"
+  role  = join("", aws_iam_role.csi-driver-role.*.name)
+}
+
+resource "aws_iam_role_policy" "csi-driver-role-policy" {
+  count = var.set_csi_driver_iam_role ? 1 : 0
+  name  = "${var.cluster_name}-policy"
+  role  = join("", aws_iam_role.csi-driver-role.*.id)
```

Comment:
I think it's safer and more intuitive to do `role = aws_iam_role.csi-driver-role.[count.index].id`.

Reply:
This won't work, because the resource `aws_iam_role.csi-driver-role` won't exist. Also, if done like this, you will look up on index 1, so 2nd instance of the resource. Or does your syntax magically handles it somehow?

Reply:
As mentioned, count.index does not work here if the roles aren't set.